### DISCUSSION TITLE
Prevent modal closing after file download

### DISCUSSION THREAD

**Discussion by underdpt** [filamentphp/filament#14084] [Help]
### Package

Form builder

### Package Version

v3

### How can we help you?

Hello,

So, I have this action that opens a modal with some help for the user. This includes downloading a file but i would like to maintain the modal open so the user can view the help while working with the downloaded file.

I'm at a point where my modal opens with a button to download the file (ideally it should download automatically) but clicking that button does nothing (the icon spins, but the code doesn't run).

This is the relevant code:

```php
Forms\Components\Actions::make([
    Forms\Components\Actions\Action::make('impresion.prueba')
        ->label('Imprimir prueba')
        ->extraModalFooterActions(function () {
            return [
                Action::make('prueba')
                    ->label('Download')
                    ->action(function (array $state, Pages\CreateEtiqueta $livewire) {
                        // This is never reached
                        Notification::make()
                            ->success()
                            ->title('File downloade')
                            ->send();

                        // This downloads the file
                        return DescargaEtiquetaPrueba::run();
                    }),
            ];
        }),
```

Can this be done?

Thanks!

**Comment by leandrocfe** [comment]
try `$livewire->halt()`

**Reply by underdpt** [reply]
Thank you,

But that doesn't work because I need to return the download, a halt before the download wouldn't trigger the download, and after the download it simply won't get to that line because of the return.

**Reply by leandrocfe** [reply]
What happens if you remove return?

**Reply by underdpt** [reply]
The file isn't downloaded at all.

**Reply by leandrocfe** [reply]
please share the whole code or provide a minimal repo on Github to reproduce this issue
